### PR TITLE
In-app announcements

### PR DESCRIPTION
**Pull request by peripheralist**
## What does this PR do and why?

Adds an in-app announcement architecture, allowing announcements to be shown _once_ to users under certain conditions. 

**Currently feature flagged (`jb.enableFeatureFlag('announcements')`). All defined announcements can be considered placeholders.**

**Architecture overview:**
- Single instance of `AnnouncementsProvider` in the app wrapper, which provides an `AnnouncementsContext` with a `setActiveId()` property
- Multiple instances of `AnnouncementsLauncher` wrapper component where needed. Launchers depend on `AnnouncementsContext.setActiveId()` and can be instantiated in multiple places to take make use of different conditional variables that may be needed to determine if an announcement should be launched. (i.e. to launch an announcement dependent on ProjectContext, we need a launcher below ProjectContext in the app tree)
- `const announcements: Announcement[]` is a list of hard-coded announcements that can be displayed. Each of these includes:
    - unique `id`
    - `conditions` array of callbacks. All conditions must return true for the announcement to be displayed
    - `content` React component to be shown in the announcement modal when activated
    - `cta` (optional) custom "OK" text and callback when clicked (e.g. "Go to settings" to check out new project tags)
    - `expire` (optional) a timestamp after which the announcement won't be shown anymore
- Once an announcement has been shown, `AnnouncementsProvider` records it in localStorage to avoid showing it to the same user again

## Screenshots or screen recordings

Mobile:
image
Desktop:
image

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

File: src/components/announcements/IntroProjectTags.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export default function IntroProjectTags() {
+  return (
+    <div>
+      <h1>Introducing project tags</h1>
```
Comment (peripheralist) [inline src/components/announcements/IntroProjectTags.tsx:4]: placeholder

File: src/components/announcements/IntroSepanaSearch.tsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export default function IntroSepanaSearch() {
```
Comment (peripheralist) [inline src/components/announcements/IntroSepanaSearch.tsx:1]: placeholder

File: src/components/common/CoreAppWrapper/CoreAppWrapper.tsx [hunk 3]
```diff
@@ -33,7 +35,11 @@ export const AppWrapper: React.FC = ({ children }) => {
               <ThemeProvider>
                 <EtherPriceProvider>
                   <ArcxProvider>
-                    <_Wrapper>{children}</_Wrapper>
+                    <AnnouncementsProvider>
```
Comment (peripheralist) [inline src/components/common/CoreAppWrapper/CoreAppWrapper.tsx:38]: there will only be one `AnnouncementsProvider` in the tree

File: src/components/common/CoreAppWrapper/CoreAppWrapper.tsx [hunk 4]
```diff
@@ -33,7 +35,11 @@ export const AppWrapper: React.FC = ({ children }) => {
...
+                      <AnnouncementLauncher>
```
Comment (peripheralist) [inline src/components/common/CoreAppWrapper/CoreAppWrapper.tsx:39]: there can be multiple `AnnouncementLaunchers` in the tree. currently there are 2:

- one here, to handle launching announcements from top level pages like the home page or /projects
- one wrapping v1 and v2 dashboards, to handle launching announcements dependent on ProjectContexts

File: src/constants/announcements.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
+import IntroProjectTags from 'components/announcements/IntroProjectTags'
+import IntroSepanaSearch from 'components/announcements/IntroSepanaSearch'
+import { Announcement } from 'models/announcement'
+import { NextRouter } from 'next/router'
+
+export const announcements: Announcement[] = [
```
Comment (wraeth-eth) [inline src/constants/announcements.ts:6]: Should we add some conditions to stop expired announcements from showing?

File: src/components/announcements/IntroProjectTags.tsx [hunk 6]
```diff
@@ -0,0 +1,10 @@
+export default function IntroProjectTags() {
```
Comment (wraeth-eth) [inline src/components/announcements/IntroProjectTags.tsx:1]: Is it worth wrapping these in trans?

**Review by wraeth-eth — APPROVED** [review]
Dope feature @peripheralist !

File: src/contexts/Announcements/AnnouncementLauncher.tsx [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+  const { projectOwnerAddress } = useContext(V2V3ProjectContext)
+  const isProjectOwner = useIsUserAddress(owner ?? projectOwnerAddress)
+  const router = useRouter()
+
+  const { setActiveId } = useContext(AnnouncementsContext)
+
+  const shouldActivateAnnouncement = useCallback(
+    (a: Announcement) => {
+      // Don't activate if expired
+      if (a.expire && a.expire > Date.now().valueOf()) return false
```
Comment (peripheralist) [inline src/contexts/Announcements/AnnouncementLauncher.tsx:29]: @wraeth-eth expired check is here

File: src/components/common/CoreAppWrapper/CoreAppWrapper.tsx (hunk 3 above)
Comment (peripheralist) [reply src/components/common/CoreAppWrapper/CoreAppWrapper.tsx:36]: it's in the launcher!

File: src/components/announcements/IntroProjectTags.tsx (hunk 6 above)
Comment (peripheralist) [reply src/components/announcements/IntroProjectTags.tsx:1]: oh yes, they definitely should be. these are placeholders for now but will wrap them before launch

**Pull request merged**